### DISCUSSION TITLE
Formula which uses git internally fails with "git: This shim is internal and must be run via brew."

### DISCUSSION THREAD

**Discussion by CAMOBAP** [Homebrew/discussions#1162] [Tap maintenance and Homebrew development] [answered]
I faced this problem during testing formula https://github.com/metanorma/homebrew-metanorma/blob/master/Formula/metanorma.rb

Unfortunatelly I missed the moment when exactly this begin to appear. And logs on CI not available anymore.

Maybe I should define some environment variable to allow me to bypass this error?

**Comment by carlocab** [comment, marked as answer]
Check the contents of your `metanorma` env script. You've filled the `PATH` with things that are only meant for internal `brew` usage.

**Reply by CAMOBAP** [reply]
@carlocab thanks a lot for your response, maybe you can suggest a way how I can add some paths for PATH just for my formula?

Thanks In advance

**Reply by CAMOBAP** [reply]
Simple `"$PATH"` instead `ENV["PATH"]` works

**Reply by gromgit** [reply]
>how I can add some paths for PATH just for my formula?

Use https://rubydoc.brew.sh/SharedEnvExtension.html#prepend_path-instance_method or https://rubydoc.brew.sh/SharedEnvExtension.html#append_path-instance_method.
